include/deal.II/matrix_free/matrix_free.h

```diff
@@ -2923,10 +2923,12 @@ namespace internal
   template <int dim, typename Number, typename VectorizedArrayType>
   struct VectorDataExchange
   {
-    // An arbitrary shift for communication to reduce the risk for accidental
+    // A shift for the MPI messages to reduce the risk for accidental
     // interaction with other open communications that a user program might
-    // set up
-    static constexpr unsigned int channel_shift = 103;
+    // set up (parallel vectors support unfinished communication). We let
+    // the other vectors use the first 20 assigned numbers and start the
+    // matrix-free communication.
+    static constexpr unsigned int channel_shift = 20;
```

Comment:
Where is this shift used?

Reply:
(I worry that somebody selecting 20 as their channel will conflict again)

Reply:
It is used for various `LinearAlgebra::distributed::Vector::update_ghost_values_start()` and `_ghost_values_end()` calls, i.e., similarly as we do for the block vector, just using `20` as a shift rather than `100`. The point is: No matter what we choose here, there is always a chance that other communications might go on at the same time. This call is in the range of allowed channels for the parallel vector, and it is unlikely that an exchange not coming from `MatrixFree` would be going on at the same time and use the same MPI tags. The only thing I can imagine, without the user explicitly trying to break it, would conflict is two `MatrixFree` object, which both have the same number of ghost entries to be exchanged, and use the same channel and communicator. That would likely be some multi-threading locally with two of those objects in some embarrasingly parallel approach. Not sure how to approach that - I think at that point we should rather aim to duplicate the MPI communicators as @peterrum suggests.

Reply:
Alright. This is not ideal but likely to work for now. I am okay with merging this PR as is.

Would it make sense to duplicate the communicator for the MatrixFree object?